hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,762 @@
+In Pass-though mode, control of the `cs_n` lines passes directly to the inter-module port, `pt_cs_ni`.
+
+### Arbitrary Length Commands
+
+Typically the length of each command is limited by the maximum number of bits in the {{< regref "COMMAND_0.TXN_CNT_0" >}}, {{< regref "COMMAND_0.RX_CNT_0" >}}, and {{< regref "COMMAND_0.TX1_CNT_0" >}} fields.
+However, a series  of SPI_HOST IP commands can be merged into one long command sequence using the {{< regref "CONFIG_0.CSAAT_0" >}} field, which leaves the `cs_n` line low after the first SPI_HOST command.
+By raising the `cs_n` line only after the last command has completed, the peripheral will interpret the resulting transmit/receive sequences as one long command.
+
+### Opportunistic Support for Back-to-back Commands
+
+There is no command queue in the SPI_HOST IP.
+However a second command can be placed into the Config/Command CDC whenever {{< regref "STATUS.READY" >}} is high, even if the previous command is still running.
+The internal state within the Config/Command CDC provides the option of scheduling a second command to execute immediately after completion of the current command.
+
+On the other hand, writing a one to {{< regref "COMMAND_0.GO_0" >}} (or any `GO` bit in this multi-register) when {{< regref "STATUS.READY" >}} is low will trigger an error condition, which must be acknowledged by software.
+
+## Data Formatting
+
+### Input and Output Byte Ordering
+
+The SPI transactions must be issued with correct bit ordering to properly communicate with a remote device.
+Based on the requirements for our chosen flash devices, this IP follows these conventions:
+- The relative significance of lines on the `sd` bus: `sd[0]` is always the most significant, followed by `sd[1]` though `sd[3]` with decreasing significance.
+- The relative significance of a sequence of bits on the same `sd` bus: higher significance bits are always transmitted earlier than (or at the same time as) any lower significance bits.
+    - For instance, when transferring a single byte in Quad mode, all four bits of the upper nibble (bits 7 through 3) are transferred in the first clock cycle and the entire lower nibble (bits 3 through 0) is transferred in the second cycle.
+
+The programming model for the IP should meanwhile make it easy to quickly program the peripheral device, with a minimum amount of byte shuffling.
+It should be intuitive to program the specific flash devices we are targeting, while following the conventions above:
+- When reading the data in and out of the {{< regref "TXDATA" >}} and {{< regref "RXDATA" >}} registers, the IP should make the most of the TL-UL bus, using 32-bit I/O instructions.
+- The SPI_HOST should make it easy to arrange transaction-data in processor memory, meaning that bytes should be sequentially transmitted in order of ascending memory address.
+  - When using 32-bit I/O instructions, this requires some knowledge of the processor byte-order.
```

Comment:
Hi @swarbrick,
   I admit I've done something confusing here, but I've moved a section from #5077 to here.

I'm also moving the last few comments on that PR. [such as this one](https://github.com/lowRISC/opentitan/pull/5077#discussion_r570865942):

> Since OpenTitan will definitely have a little-endian host processor (Ibex), I wonder whether this needs rewording a bit. I can totally see the benefit of a "byte re-ordering mode" for when you have a pesky peripheral that wants stuff in the other order from what we've stored, but I'm not sure it makes sense to talk about processor byte-order. Or have I misunderstood something?